algorithms/scaling/outlier_rejection.py

```diff
@@ -71,6 +72,27 @@ def reject_outliers(reflection_table, experiment, method="standard", zmax=6.0):
     return reflection_table
 
 
+def determine_Esq_outlier_index_arrays(Ih_table, experiment, emax=4.0):
+    # first calculate normalised intensities and set in the Ih_table.
+    if not emax:
+        return [flex.size_t() for _ in range(Ih_table.n_datasets)]
+    intensities = Ih_table.as_miller_array(experiment.crystal.get_unit_cell())
+    normalised_intensities = quasi_normalisation(intensities)
+
+    sel = normalised_intensities.data() > (emax ** 2)
+    logger.info(
+        f"{sel.count(True)} outliers identified from normalised intensity analysis (E\xb2 > {(emax ** 2)})"
+    )
+    outlier_indices = Ih_table.Ih_table_blocks[0].Ih_table["loc_indices"].select(sel)
+    datasets = Ih_table.Ih_table_blocks[0].Ih_table["dataset_id"].select(sel)
+    if Ih_table.n_datasets == 1:
+        return [outlier_indices]
```

Comment:
Is there a real reason for separately handle the case `n_datasets == 1`? I can't imagine it results in a quantifiable performance benefit, and adds code complexity.

Reply:
I've now moved this line up by one, and in the case of `n_datasets = 1` it now saves doing two selections, one of which is on the whole dataset, so I'd like to keep it this way as it will be called multiple times per job.